keepercommander/loginv3.py

```diff
@@ -506,13 +511,39 @@ def handleTwoFactor(params: KeeperParams, encryptedLoginToken, login_resp):
             elif channel_type == 'TWO_FA_CT_SMS':
                 prompt_str = prompt_str + " (use code sent to)"
 
-            otp_code: str = input('\n' + prompt_str + ': ')
+            otp_code = ''
+            show_duration = True
+            mfa_pattern = re.compile(r'2fa_duration\s*=\s*(.+)', re.IGNORECASE)
+            while not otp_code:
+                if show_duration:
+                    show_duration = False
+                    prompt_exp = '\n2FA Code Duration: {0}.\nTo change duration: 2fa_duration=login|30_days|forever' \
+                        .format('Require Every Login' if mfa_expiration == proto.TWO_FA_EXP_IMMEDIATELY else
+                                'Save on this Device Forever' if mfa_expiration == proto.TWO_FA_EXP_NEVER else
+                                'Ask Every 30 days')
+                    print(prompt_exp)
+
+                answer = input('\n' + prompt_str + ': ')
```

Comment:
Why does the user have to enter 2fa_duration=whatever?  Are we planning for more options here someday?

Reply:
This "2fa_duration=whatever" is optional. User does not have to enter it. The default value will be assumed if  2fa_duration is omitted. Yes, this area may be extended with additional options.